### PR TITLE
Implement generic from and to JSON support

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

**Comment by tribal-tec** [comment]
FYI @chevtche

File: zerobuf/Schema.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * The type identifier of the object.
+     * @sa ZeroBuf::getZerobufType()
+     */
+    const servus::uint128_t type;
+
+    /** Access enums for the tuple per field in the object.*/
+    enum EntryEnum
+    {
+        ENTRY_KEY,        //!< the name of the field
```
Comment (eile) [inline zerobuf/Schema.h]: ENTRY_NAME then?

File: zerobuf/Vector.h [hunk 2]
```diff
@@ -14,50 +14,54 @@ namespace zerobuf
...
-    typedef BaseVector< Allocator, T, I > Super;
+    typedef BaseVector< Allocator, T > Super;
 
 public:
-    Vector( Allocator* alloc );
+    /**
+     * @param alloc The parent allocator that contains the data.
+     * @param index Index of the vector in the parent allocator dynamic storage
+     */
+    Vector( Allocator* alloc, const size_t index );
```
Comment (eile) [inline zerobuf/Vector.h]: rm const

**Comment by tribal-tec** [comment]
Updated

**Pull request merged**